spade_llm/core/models.py

```diff
@@ -21,14 +21,22 @@ def inject_env(val: str):
             return val
 
     @staticmethod
-    def inject_env_dict(keys: list[str], conf: dict[str, Any]):
+    def inject_env_dict(keys: list[str], conf: dict[str, Any], extra_params: dict[str, Any] = None):
         """
         For each key in keys list check if it is mentioned in dict and applies inject_env to its value.
         Used to substitute credentials from environment variables.
+        You may provide extra arguments into config via extra_params.
         """
         for key in keys:
             if key in conf:
-                conf[key] = CredentialsUtils.inject_env(conf[key])
+                if isinstance(conf[key], SecretStr):
+                    conf[key] = SecretStr(CredentialsUtils.inject_env(conf[key].get_secret_value()))
```

Comment:
Did not get it - why do we need inject twice and how ca we get SecretStr from inject?